server/controllers/MiscController.js

```diff
@@ -52,43 +50,29 @@ class MiscController {
       return res.status(500).send(`Invalid post data`)
     }
 
-    // For setting permissions recursively
-    let outputDirectory = ''
-    let firstDirPath = ''
-
-    if (library.isPodcast) { // Podcasts only in 1 folder
-      outputDirectory = Path.join(folder.fullPath, title)
-      firstDirPath = outputDirectory
-    } else {
-      firstDirPath = Path.join(folder.fullPath, author)
-      if (series && author) {
-        outputDirectory = Path.join(folder.fullPath, author, series, title)
-      } else if (author) {
-        outputDirectory = Path.join(folder.fullPath, author, title)
-      } else {
-        outputDirectory = Path.join(folder.fullPath, title)
-      }
-    }
-
-    if (await fs.pathExists(outputDirectory)) {
-      Logger.error(`[Server] Upload directory "${outputDirectory}" already exists`)
-      return res.status(500).send(`Directory "${outputDirectory}" already exists`)
-    }
+    // Podcasts should only be one folder deep
+    const outputDirectoryParts = library.isPodcast ? [title] : [author, series, title]
+    // `.filter(Boolean)` to strip out all the potentially missing details (eg: `author`)
+    // before sanitizing all the directory parts to remove illegal chars and finally prepending
+    // the base folder path
+    const cleanedOutputDirectoryParts = outputDirectoryParts.filter(Boolean).map(part => sanitizeFilename(part))
+    const outputDirectory = Path.join(...[folder.fullPath, ...cleanedOutputDirectoryParts])
 
     await fs.ensureDir(outputDirectory)
 
     Logger.info(`Uploading ${files.length} files to`, outputDirectory)
 
-    for (let i = 0; i < files.length; i++) {
-      var file = files[i]
+    for (const file of files) {
+      const path = Path.join(outputDirectory, sanitizeFilename(file.name))
 
-      var path = Path.join(outputDirectory, file.name)
-      await file.mv(path).then(() => {
-        return true
-      }).catch((error) => {
-        Logger.error('Failed to move file', path, error)
-        return false
-      })
+      await file.mv(path)
+        .then(() => {
+          return true
+        })
+        .catch((error) => {
+          Logger.error('Failed to move file', path, error)
+          return false
+        })
```

Comment:
okay, I cannot figure this out and I spent way too much time trying to reason around it. `.mv` returns a promise, so why can't I `await` it alone? Why does it need to be chained to another promise? Why can't I use `try`/`catch` here? I dug into the code of `express-fileupload` and the way this promise is called is so contrived that I truly couldn't understand it.  I could do something like this:

```js
await new Promise((resolve, reject) => {
  file.mv(path, (err) => {
    if (err) reject(err)
    resolve()
  })
})
```

But that seems _very_ overkill just to be able to interact with it like a normal promise? I am _very_ tired so maybe I'm missing something obvious, but for now this stays as it was